File: projects/poker/julia/card.jl
mutable struct Card
	rank::String
	suit::String
end

function toString(card::Card)
	return card.rank * card.suit
end

function rankValue(card::Card)
	if card.rank == "A"  # A is the highest so it returns 14 rather than 1
		return 14
	elseif card.rank == "K"
		return 13
	elseif card.rank == "Q"
		return 12
	elseif card.rank == "J"
		return 11
	elseif card.rank == "10"
		return 10
	else
		return parse(Int, card.rank)
	end
end

function suitValue(card::Card)
	if card.suit == "D"
		return 1
	elseif card.suit == "C"
		return 2
	elseif card.suit == "H"
		return 3
	elseif card.suit == "S"
		return 4
	end

	return 0
end

function equals(card1::Card, otherCard::Card)
	return card1.rank == otherCard.rank && card1.suit == otherCard.suit
end

File: projects/poker/c#/Card.cs
using System;

public class Card
{
	public string rank;
	public string suit;

	public Card()
	{
		this.rank = "";
		this.suit = "";
	}

	public Card(string rank, string suit)
	{
		this.rank = rank;
		this.suit = suit;
	}

	public string toString()
	{
		return rank + suit;
	}

	public int rankValue()
	{
		if (rank == "A") // A is the highest so it returns 14 rather than 1
			return 14;
		if (rank == "K")
			return 13;
		if (rank == "Q")
			return 12;
		if (rank == "J")
			return 11;
		if (rank == "10")
			return 10;
		return int.Parse(rank);
	}

	public int suitValue()
	{
		if (suit == "D")
			return 1;
		if (suit == "C")
			return 2;
		if (suit == "H")
			return 3;
		if (suit == "S")
			return 4;
		return 0;
	}

	public bool equals(Card otherCard)
	{
		return this.rank == otherCard.rank && this.suit == otherCard.suit;
	}
}

